sonic_platform_base/module_base.py

```diff
@@ -77,6 +89,83 @@ def get_system_eeprom_info(self):
         """
         raise NotImplementedError
 
+    def get_name(self):
+        """
+        Retrieves the name of the module prefixed by CONTROL-CARD, LINE-CARD,
+        FABRIC-CARD
+
+        Returns:
+            string: A string providing the name of the card prefixed by one of the
+            MODULE_TYPE_CONTROL, MODULE_TYPE_LINE, MODULE_TYPE_FABRIC followed by
+            an index.
+            Ex. A Chassis having 1 control-card, 4 line-cards and 6 fabric-cards
+            can provide names CONTROL-CARD1, LINE-CARD1 to LINE-CARD4,
+            FABRIC-CARD1 to FABRIC-CARD6
+        """
+        raise NotImplementedError
+
+    def get_description(self):
```

Comment:
do we need set_description so that user can change it?

Reply:
Did not see a need for this at this point. This comes from the platform vendor.